### PR TITLE
add re-usable navigation bar

### PR DESCRIPTION
**Pull request by ibsa-ejeta**
### What this PR does (required):
- Adds a navigation bar to the project
- This navigation bar can be used for another project

### Screenshots / Videos (required):

ticket15first

ticket15secon

ticket15third

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- Clicking on the options will take you to logout
- Routing functionality is not integrated

### CODE REVIEW

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:37]: Especially with functional components & hooks, we should try to use a single `useState` hook for each part of the state (e.g. one for `mobileView` and one for `drawerOpen`) unless we have a good reason not to. I think these would definitely be better separated.

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:73]: We probably don't need to define this inside of the `Navbar` component - it's better to separate each of these functions into their own components. If you want to you, can pass `className`s (as well as other props) to these in order avoid re-using the `useStyles` hook within each component too. No need to separate the files though!

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:75]: This as well!

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:115]: This too!

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:128]: Could this be missing the logo content?

File: client/src/components/Navbar/Navbar.tsx
Comment (ugur-akin) [inline client/src/components/Navbar/Navbar.tsx:130]: This could be moved outside as well.

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (ugur-akin) [inline client/src/pages/Dashboard/Dashboard.tsx:39]: Since `loggedInUser` is provided with a context, there is probably no need to pass this down. We can use the `useAuth` hook as needed within `Navbar` or its sub-components.

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (ugur-akin) [inline client/src/pages/Dashboard/Dashboard.tsx:38]: I believe these are routes passed down to the navbar, in that case I would recommend using an [enum-like object](https://www.sohamkamani.com/javascript/enums/#enumerations-with-objects) to define routes that can be used project-wide.

**Review by ugur-akin — COMMENTED** [review]
Some minor organization issues, should be good to merge after some refactoring!

File: client/src/components/Navbar/Navbar.tsx
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx:34]: I second Ugur's comment below. Also, maybe consider giving the state variable a more descriptive name?

File: client/src/components/Navbar/Navbar.tsx
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx:52]: 👍🏽

File: client/src/components/Navbar/Navbar.tsx
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx:154]: MUI `<Box>` components are divs by default, we don't need to specify it here

**Review by rajivtitus — COMMENTED** [review]
Good effort! Please address comments when possible. Thanks!

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: created two separate states

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: created new component name DisplayMobile

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: created new component name GetDrawerChoices

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: removed

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: created new component named GetMenuButtons

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (ibsa-ejeta) [reply client/src/pages/Dashboard/Dashboard.tsx]: used useAuth in GetMenuButton component and removed loggedInUser

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (ibsa-ejeta) [reply client/src/pages/Dashboard/Dashboard.tsx]: followed the standard Enumerations with objects

File: client/src/components/Navbar/Navbar.tsx
Comment (ibsa-ejeta) [reply client/src/components/Navbar/Navbar.tsx]: given descriptive name to both states

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes.

Just a minor comment, please address before merging. Thank you!

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (rajivtitus) [inline client/src/pages/Dashboard/Dashboard.tsx:33]: Please use enums to define routes, see article below for reference.

https://betterprogramming.pub/the-best-way-to-manage-routes-in-a-react-project-with-typescript-c4e8d4422d64

Comment (ibsa-ejeta) [reply client/src/pages/Dashboard/Dashboard.tsx:33]: fixed the naming as per the standard
created 5 routes and 5 strings which can be modified and re-used

**Pull request merged**
